server/src/server.ts

```diff
@@ -100,6 +102,23 @@ async function validateTextDocument(textDocument: TextDocument): Promise<void> {
 	connection.sendDiagnostics({ uri: textDocument.uri, diagnostics }); 
 }
 
+connection.onDocumentFormatting(
+	(params: DocumentFormattingParams): TextEdit[] => {
+		let reader = ion.makeReader(documentText);
+		let writer = ion.makePrettyWriter();
+		writer.writeValues(reader);
+		writer.getBytes();
+		writer.close();
+
+		documentText = String.fromCharCode.apply(null, Array.from(writer.getBytes()));
+
+		let textEdits: TextEdit[] = [];
+		textEdits.push(TextEdit.replace(Range.create(0, 0, 9999, 9999), documentText));
```

Comment:
What happens if the file has more than 9999 lines? Is there a way we can get the contents and replace it with the formatted text without the need to specify the range?

Reply:
Can I replace 'Range.create(0, 0, 9999, 9999)' into 'Range.create(0, 0, document.length + 1, 0)'? The third parameter is the endLine.

Reply:
A solution that would not break because of a magic number is better.